src/Toml/BiMap.hs

```diff
@@ -100,52 +104,52 @@ _Right = invert $ prism (either (const Nothing) Just) Right
 -- | Creates prism for 'AnyValue'.
 mkAnyValueBiMap :: (forall t . Value t -> Maybe a)
                 -> (a -> Value tag)
-                -> BiMap AnyValue a
+                -> BiMap a AnyValue
 mkAnyValueBiMap matchValue toValue =
-    prism (\(AnyValue value) -> matchValue value) (AnyValue . toValue)
+    BiMap (Just . AnyValue . toValue) (\(AnyValue value) -> matchValue value)
+
+-- | Allows to match against given 'Value' using provided prism for 'AnyValue'.
+matchValueBackward :: BiMap a AnyValue -> Value t -> Maybe a
+matchValueBackward = liftMatch . backward
 
--- | Creates prism for 'Text' to 'AnyValue' bimap with custom functions
-_TextBy :: (Text -> Maybe a) -> (a -> Text) -> BiMap AnyValue a
+-- | Creates prism for 'Text' to 'AnyValue' with custom functions
+_TextBy :: (Text -> Maybe a) -> (a -> Text) -> BiMap a AnyValue
```

Comment:
Let's also swap order of arguments for `_TextBy` to be more consistent with the `prism` combinator